internal/controllers/machineset/machineset_controller.go

```diff
@@ -1203,7 +1203,7 @@ func (r *Reconciler) reconcileV1Beta1Status(ctx context.Context, s *scope) error
 
 		if noderefutil.IsNodeReady(node) {
 			readyReplicasCount++
-			if noderefutil.IsNodeAvailable(node, ms.Spec.MinReadySeconds, metav1.Now()) {
+			if noderefutil.IsNodeAvailable(node, ptr.Deref(machine.Spec.MinReadySeconds, 0), metav1.Now()) {
```

Comment:
I think we're never setting MinReadySeconds on the Machine. This should be done in computeDesiredMachine (Please also ensure we have test coverage for that in the computeDesiredMachine tests)

Reply:
Also in syncMachines, (this field should be propagated in place)

Reply:
I assume you are referring to the Machine with DeletionTimestamp case in ~ l.510 (I agree)